google/cloud/bigquery/table.py

```diff
@@ -2942,6 +2953,123 @@ def __repr__(self):
         return "TimePartitioning({})".format(",".join(key_vals))
 
 
+class PrimaryKey:
+    """Represents the primary key constraint on a table's columns.
+
+    Args:
+        columns: The columns that are composed of the primary key constraint.
+    """
+
+    def __init__(self, columns: List[str]):
+        self.columns = columns
+
+    def __eq__(self, other):
+        if not isinstance(other, PrimaryKey):
+            raise NotImplementedError
```

Comment:
We have several areas of code that coverage does not believe are being tested. Those lines of code are:

* 2968
* 2985-2987
* 3019

For line 2968, it does not appear that we have a test that specifically exercises the `raise` statement by trying to compare an input value that is not a PrimaryKey.

Reply:
Yep, I missed the coverage part 😅  Thank you for pointing that out. I've added missing coverage in commit https://github.com/googleapis/python-bigquery/pull/1755/commits/29d123809ea60835c028eea27e4f3f005538d083